File: scalaapi/src/com/lightform/_2auth/scalaapi/payloads/requests/AuthorizationRequest.scala
package com.lightform._2auth.scalaapi.payloads.requests

import com.lightform._2auth.javaapi.interfaces
import scala.jdk.CollectionConverters._
import scala.jdk.OptionConverters._

case class AuthorizationRequest(
    response_type: String,
    client_id: String,
    redirect_uri: Option[String],
    scope: Set[String],
    state: Option[String]
) extends interfaces.AuthorizationRequest {
  def getResponseType = response_type
  def getClientId     = client_id
  def getRedirectUri  = redirect_uri.toJava
  def getScope        = scope.asJava
  def getState        = state.toJava
}

File: 2auth/src/com/lightform/_2auth/javaapi/interfaces/AuthorizationRequest.java
package com.lightform._2auth.javaapi.interfaces;

import java.util.Optional;
import java.util.Set;

public interface AuthorizationRequest {
	String getResponseType();
	String getClientId();
	Optional<String> getRedirectUri();
	Set<String> getScope();
	Optional<String> getState();
}
